internal/cli/k8s_bootstrap.go

```diff
@@ -0,0 +1,366 @@
+	_, err = client.SetServerConfig(ctx, &pb.SetServerConfigRequest{
+		Config: &pb.ServerConfig{
+			AdvertiseAddrs: []*pb.ServerConfig_AdvertiseAddr{
+				{
+					Addr:          advertiseAddr,
+					Tls:           c.flagAdvertiseTLS,
+					TlsSkipVerify: c.flagAdvertiseTLSSkipVerify,
+				},
+			},
+			Platform: "kubernetes",
+		},
+	})
+	if err != nil {
+		c.ui.Output(
+			"Error setting server configuration: %s",
+			clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	// Get our runner token
+	log.Info("generating login token for runner")
+	tokenResp, err := client.GenerateLoginToken(c.Ctx, &pb.LoginTokenRequest{
+		Duration: "", // Never expire for the static runner
+	})
+	if err != nil {
+		c.project.UI.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+	runnerTokenB64 := base64.StdEncoding.EncodeToString([]byte(tokenResp.Token))
+
+	// Persist our root token
+	log.Info("persisting root token", "secret", c.flagRootTokenSecret)
+	_, err = secretClient.Patch(ctx, c.flagRootTokenSecret, types.JSONPatchType, []byte(
+		fmt.Sprintf(`[{"op":"replace", "path": "/data/token", "value": "%s"}]`, bootstrapTokenB64)),
+		metav1.PatchOptions{})
+	if err != nil {
+		c.ui.Output(
+			"Error patching root token secret: %s",
+			clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	// Persist our runner token
+	log.Info("persisting runner token", "secret", c.flagRunnerTokenSecret)
+	_, err = secretClient.Patch(ctx, c.flagRunnerTokenSecret, types.JSONPatchType, []byte(
+		fmt.Sprintf(`[{"op":"replace", "path": "/data/token", "value": "%s"}]`, runnerTokenB64)),
+		metav1.PatchOptions{})
+	if err != nil {
+		c.ui.Output(
+			"Error patching runner token secret: %s",
+			clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	// Configure our on-demand runner
+	log.Info("storing on-demand runner configuration for Kubernetes")
+	_, err = client.UpsertOnDemandRunnerConfig(ctx, &pb.UpsertOnDemandRunnerConfigRequest{
+		Config: &pb.OnDemandRunnerConfig{
+			OciUrl:     c.flagODRImage,
+			PluginType: "kubernetes",
+			PluginConfig: []byte(fmt.Sprintf(`{
+	"service_account": "%s",
+	"image_pull_policy": "%s"
+}`, c.flagODRServiceAccount, c.flagODRImagePullPolicy)),
+			ConfigFormat: pb.Project_JSON,
+			Default:      true,
+		},
+	})
+
+	log.Info("bootstrap complete")
+	return 0
+}
+
+func (c *K8SBootstrapCommand) Flags() *flag.Sets {
+	return c.flagSet(flagSetConnection, func(set *flag.Sets) {
+		f := set.NewSet("Command Options")
+		f.StringVar(&flag.StringVar{
+			Name:   "root-token-secret",
+			Target: &c.flagRootTokenSecret,
+			Usage:  "The name of the Kubernetes secret to write the root token to.",
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:   "runner-token-secret",
+			Target: &c.flagRunnerTokenSecret,
+			Usage:  "The name of the Kubernetes secret to write the runner token to.",
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:   "odr-image",
```

Comment:
Nit: should we prefix these with `k8s-` to be consistent with the normal [K8s installation flags](https://www.waypointproject.io/commands/install#kubernetes-options)? Maybe we aren't concerned so much because this is an internal thing, so I'm fine with either way.

Reply:
Makes sense thanks 👍